dapp/src/components/buySell/SellWidget.js

```diff
@@ -221,10 +222,46 @@ const SellWidget = ({
         result = await vaultContract.redeemAll(minStableCoinsReceivedBN, {
           gasLimit,
         })
-        storeTransaction(result, `redeem`, returnedCoins, coinData)
+        receipt = await rpcProvider.waitForTransaction(result.hash)
+        const data = {
+          usdt: 0,
+          dai: 0,
+          usdc: 0,
+          ousd: 0,
+        }
+        await Promise.all(
+          receipt.logs.map(async (log) => {
+            if (
+              log.address.toLowerCase() === addresses.mainnet.USDT.toLowerCase()
+            ) {
+              const value = ethers.BigNumber.from(log.data)
+              const decimals = await usdtContract.decimals()
+              data.usdt = parseFloat(value.toString()) / 10 ** decimals
+            } else if (
+              log.address.toLowerCase() === addresses.mainnet.USDC.toLowerCase()
```

Comment:
A lot of the code after the `if` condition is shared between the different parts of `if/else` clauses. This could probably be shortened with smth like: 
```
receipt.logs.map(async (log) => { 
  const stableCoinMap={
     addresses.mainnet.DAI.toLowerCase(): {
         name: 'dai',
         decimals: 18,
     },
     addresses.mainnet.USDT.toLowerCase(): ...
  }

  const stableCoin = stableCoinMap[log.address.toLowerCase()]
  if (!stableCoin) {
    // we don't know which stable coin it is
  }
 
  const value = ethers.BigNumber.from(log.data) 
  const decimals = stableCoin.decimals
  data[stableCoin.name] = parseFloat(value.toString()) / 10 ** decimals
}
```